File: api/request/response.go
package request

type Error struct {
	Code    string      `json:"code"`
	Message string      `json:"message"`
	Target  interface{} `json:"target,omitempty"`
	status  int
}

type Response struct {
	Error   interface{} `json:"error"`
	Data    interface{} `json:"data"`
	err     error
	code    int
	headers map[string]string
}

File: src/Volunteasy.Api/Response/Response.cs
using System.Text.Json.Serialization;

namespace Volunteasy.Api.Response;

public record Response
{
    [JsonIgnore]
    public HttpContext Context { get; init; } = null!;

    public int Status => Context.Response.StatusCode;

    public bool Success => Context.Response.StatusCode is >= 200 and <= 299;
    
    public string CorrelationId => Context.TraceIdentifier;
    
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? Reason { get; init; }

    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public string? Message { get; init; }
    
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public object? Data { get; init; }
}